governator-api/src/main/java/com/netflix/governator/event/ApplicationEventDispatcher.java

```diff
@@ -0,0 +1,19 @@
+package com.netflix.governator.event;
+
+import java.lang.reflect.Method;
+
+/**
+ * Interface to publishing {@link ApplicationEvent}s as well as programmatically registering
+ * {@link ApplicationEventListener}s.
+ */
+public interface ApplicationEventDispatcher {
+
+    <T extends ApplicationEvent> void registerListener(Class<T> eventType, ApplicationEventListener<T> eventListener);
```

Comment:
No strong feelings either way. I can't imagine when I'd use it, but guava certainly supports it. Seems like an easy add just in case someone needs it.